pmd-java/src/test/java/net/sourceforge/pmd/lang/java/oom/DataStructureTest.java

```diff
@@ -30,13 +46,39 @@
  */
 public class DataStructureTest extends ParserTst {
 
+    MetricKey<ClassMetric> classMetricKey = new MetricKey<ClassMetric>() {
+        @Override
+        public String name() {
+            return null;
+        }
+
+
+        @Override
+        public ClassMetric getCalculator() {
+            return new RandomMetric();
```

Comment:
Nice idea with the RandomMetric for tests.
I like it, that you can now define such metrics adhoc for testing!